wasm/jit/jit_amd64.go

```diff
@@ -0,0 +1,1156 @@
+		case *wazeroir.OperationOr:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationXor:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationShl:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationShr:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationRotl:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationRotr:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationAbs:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationNeg:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationCeil:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationFloor:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationTrunc:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationNearest:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationSqrt:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationMin:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationMax:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationCopysign:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationI32WrapFromI64:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationITruncFromF:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationFConvertFromI:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationF32DemoteFromF64:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationF64PromoteFromF32:
+		case *wazeroir.OperationI32ReinterpretFromF32,
+			*wazeroir.OperationI64ReinterpretFromF64,
+			*wazeroir.OperationF32ReinterpretFromI32,
+			*wazeroir.OperationF64ReinterpretFromI64:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationExtend:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		default:
+			return nil, fmt.Errorf("unreachable: a bug in JIT compiler")
+		}
+	}
+
+	if len(builder.onLabelStartCallbacks) > 0 {
+		keys := make([]string, 0, len(builder.onLabelStartCallbacks))
+		for key := range builder.onLabelStartCallbacks {
+			keys = append(keys, key)
+		}
+		return nil, fmt.Errorf("labels are not defined: %s", strings.Join(keys, ","))
```

Comment:
does this mean they are not supported? I don't understand why we are checking this externally.. should `.assemble()` return this error instead?

Reply:
delete this check as this is in any way checked in wazeroir compiler https://github.com/tetratelabs/wazero/pull/60/commits/3d35deafdc1700724a3861ff502135cfd9713518